README.md

```diff
@@ -91,8 +91,96 @@ scripts, which you can copy to your current working directory by running
 [the pyd docs](https://pyd.readthedocs.io/en/latest/dub.html) for more
 information.
 
+# .NET Support
+
+Autowrap can also generate C# bindings to D libraries. Generating bindings for C# is a two-step process. First we must generate a C# compatible C-interface for the D library, then we build and run a version of the library that emits the corresponding C# interface code.
+
+### What Works:
+
+* Primitive Types
+  * string
+  * wstring
+  * dstring
+  * byte
+  * ubyte
+  * short
+  * ushort
+  * int
+  * uint
+  * long
+  * ulong
+  * float
+  * double
+* Structs
+  * Fields
+  * Properties
+  * Methods
+* Classes
+  * Constructors
+  * Fields
+  * Properties
+  * Methods
+
+1-Dimensional Ranges are implemented but only lightly tested and may not work in all cases. Higher dimensional ranges are not supported.
+
+## Generating .NET Interfaces
+
+To generate the C-interface use the following example replacing "csharp" with the dub name of your library (ex: csharp builds to libcsharp&#46;so) and replace "csharp.library" with the name of the module you want to wrap.
 
-### Our sponsors
+```d
+import csharp.library;
+import autowrap.csharp.boilerplate;
+import autowrap.reflection;
+
+mixin(
+    wrapCSharp("csharp",
+        Modules(
+            Module("csharp.library")
+        )
+    )
+);
+```
+
+To generate the C# interface use the writeCSharpFile method as follows to dump the output to a file of your choice.
+
+```d
+string code = writeCSharpFile!(Module("csharp.library"))("csharp", "Autowrap.CSharp");
+```
+
+The template parameter is a list of module to wrap, the first parameter is the dub library name, and the second parameter is the root namespace of the C# project.
```

Comment:
Conventions like these are hard to remember and harder to get right. Please use types to disambiguate the parameters.